src/utils/components/CloudinitModal/utils/cloudinit-utils.ts

```diff
@@ -75,27 +75,17 @@ export const convertYAMLToNetworkDataObject = (networkData: string): CloudInitNe
 };
 
 export const convertNetworkDataObjectToYAML = (networkData: CloudInitNetworkData): string => {
-  const hasValue = !isEmpty(networkData?.name || networkData?.address || networkData?.gateway);
+  const hasValue = !isEmpty(networkData?.name) || !isEmpty(networkData?.addresses);
   try {
     return hasValue
       ? dump({
-          network: {
-            config: [
-              {
-                name: networkData?.name,
-                subnets: [
-                  {
-                    address: (networkData?.address || '')?.replace(/\s/g, '').split(','),
-                    gateway: networkData?.gateway,
-                    type: 'static',
-                  },
-                ],
-                type: 'physical',
-              },
-            ],
-            version: '1',
+          ethernets: {
+            [networkData?.name || '']: {
```

Comment:
i dont htink u need || "" as u are checking in hasValue that it isn't empty

Reply:
I'm checking if either name isn't empty or the addresses which means that the other can be empty, in that case I do need the `|| ''`